tests/Umbraco.Tests.Integration/Umbraco.Core/Services/UserServiceCrudTests.Update.cs

```diff
@@ -242,4 +242,143 @@ public async Task Iso_Code_Is_Validated(string isoCode, bool shouldSucceed)
         Assert.IsNotNull(updatedUser);
         Assert.AreEqual(isoCode, updatedUser.Language);
     }
+
+    [Test]
+    public async Task Can_Assign_User_Start_Nodes()
+    {
+        var contentService = GetRequiredService<IContentService>();
+        var mediaService = GetRequiredService<IMediaService>();
+        var contentStartNode = contentService.GetRootContent().First();
+        var mediaStartNode = mediaService.CreateMediaWithIdentity("test", -1, "Image");
+
+        var userService = CreateUserService(securitySettings: new SecuritySettings { UsernameIsEmail = false });
+
+        var (updateModel, createdUser) = await CreateUserForUpdate(userService);
+        updateModel.ContentStartNodeKeys = new HashSet<Guid> { contentStartNode.Key };
+        updateModel.MediaStartNodeKeys = new HashSet<Guid> { mediaStartNode.Key };
+
+        var result = await userService.UpdateAsync(Constants.Security.SuperUserKey, updateModel);
+
+        Assert.IsTrue(result.Success);
+        var updatedUser = await userService.GetAsync(createdUser.Key);
+
+        Assert.IsNotNull(updatedUser);
+        Assert.IsNotNull(updatedUser.StartContentIds);
+        Assert.AreEqual(1, updatedUser.StartContentIds.Length);
+        Assert.AreEqual(contentStartNode.Id, updatedUser.StartContentIds.First());
+        Assert.IsNotNull(updatedUser.StartMediaIds);
+        Assert.AreEqual(1, updatedUser.StartMediaIds.Length);
+        Assert.AreEqual(mediaStartNode.Id, updatedUser.StartMediaIds.First());
```

Comment:
You are asserting 2 scenarios here. Arguable 3, but the first bit is a guard assert against "normal" save operation results
- do assigned content start nodes get persisted
- do assigned media start nodes get persisted

There should be 2 additional tests as you need to prove that individual changes are persisted.
You could also add TestCases to check all 3 permutations.

Reply:
Might not be prudent to address this at this time but we should at least be aware that this test is not ideal and create a todo/task for it.